pkg/nodes/enricher/enricher_impl.go

```diff
@@ -49,7 +49,22 @@ func (e *enricherImpl) RemoveNodeIntegration(id string) {
 
 // EnrichNode enriches a node with the integration set present.
 func (e *enricherImpl) EnrichNode(node *storage.Node) error {
+	nodeNoteSet := make(map[storage.Node_Note]struct{}, len(node.GetNotes()))
+	for _, note := range node.GetNotes() {
+		nodeNoteSet[note] = struct{}{}
+	}
+
 	err := e.enrichWithScan(node)
+	if err != nil {
+		nodeNoteSet[storage.Node_MISSING_SCAN_DATA] = struct{}{}
+	} else {
+		delete(nodeNoteSet, storage.Node_MISSING_SCAN_DATA)
+	}
```

Comment:
And here (part 2) you:

```go
if err != nil {
	node.Notes = append(node.Notes, storage.Node_MISSING_SCAN_DATA)
}
```

Reply:
Also, does it worth adding logging to capture the error?

Reply:
The errors are logged by the callers of `EnrichNode`